doozerlib/cli/release_gen_payload.py

```diff
@@ -225,6 +225,10 @@ def build_dest_name(tag_name):
                 if tag_name not in mirroring[key]:
 
                     if 'cli' not in mirroring[key]:
+                        if runtime.group_config.public_upstreams and not private:
+                            # Exclude any dummy image updates for public namespaces if cli is embargoed.
+                            # If "cli" image is also missing in *-priv namespaces, the error will be raised later when attempting to update a dummy image in *-priv namespaces.
+                            break
```

Comment:
This doesn't look like the right location for this logic - 
```
if 'cli' not in mirroring[key]:
```
is just a check to make sure there *is* a dummy `cli` image to use in the replacement. The code here will only have an effect if there isn't one.

I think, after confirming that there is a `cli` image, we want to check here whether the `cli` image is ok to sync, and if it's not, `continue` here (just omit `tag_name` entirely; usually this should leave the previous dummy image in place).

Reply:
Well, I think the logic here has become a little hard to read (maybe we should rewrite that part). Let me explain a little more:
- The logic here is to append extra tags using dummy images to non-x86_64 imagestreams. Normal images are already processed.
- If `cli` is embargoed, it will be missing in any public imagestreams (`x86_64`, `ppc64le`, `s390x`). When we attempt to omit a dummy for first time, `'cli' not in mirroring[key]` test (where key is one of `x86_64`, `ppc64le`, or `s390x`) is evaluated to `True`. In this case, all succeeding tests will always be `True`. `break` should have the same effects as `continue` but have better performance.
- For private imagestreams (`x86_64-priv`, `ppc64le-priv`, `s390x-priv`), if `cli` is missing, it will raise an error.